src/server/routes/sign.rs

```diff
@@ -69,18 +79,16 @@ impl UnsignedBody {
         Ok(key_manager.verify_bundler_signature(&signature_data, &decoded_signature))
     }
 
-    // FIXME: needs proper error type
     pub async fn sign<KeyManager>(&self, key_manager: &KeyManager) -> Result<String, ()>
     where
         KeyManager: key_manager::KeyManager,
     {
-        let tx_id = self.id.as_bytes().to_vec();
         let bundler_address = key_manager.bundler_address().to_string();
 
         let signature_data = deep_hash(DeepHashChunk::Chunks(vec![
             DeepHashChunk::Chunk(VALIDATOR_AS_BUFFER.into()),
             DeepHashChunk::Chunk(ONE_AS_BUFFER.into()),
-            DeepHashChunk::Chunk(tx_id.into()),
+            DeepHashChunk::Chunk(self.id.as_bytes().to_owned().into()),
             DeepHashChunk::Chunk(bundler_address.into()),
```

Comment:
@joshbenaron is this correct data for validator signature of the received data or should we include other fields too?